src/storage.ts

```diff
@@ -68,7 +73,7 @@ export default class LlamaStorage {
     }));
   }
 
-  async add(file: Blob, metadata: FileMetadata): Promise<FileRecord> {
+  add(file: Blob, metadata: FileMetadata): FileRecordReady {
```

Comment:
This strikes me as a rather odd API from a developer ergonomics point of view. Looking at the (rough) UX we're looking for, presumably we'd want to keep the "store" button on the adding content flow (#20) spinning until adding the content is complete, after which it should be immediately visible in the rest of the UI. Based on that, it seems to me that _always_ waiting is the right thing to do?

Reply:
fiiiiiine